src/App.js

```diff
@@ -1,25 +1,58 @@
 import React from 'react';
+import { Clock } from './components/Clock';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
+class App extends React.Component {
+  state = {
+    clockName: 0,
+    isClockVisible: true,
+  }
+
+  showClock = () => this.setState({ isClockVisible: true });
+
+  hideClock = () => this.setState({ isClockVisible: false });
+
+  generateName = () => this.setState(
+    { clockName: Math.ceil(Math.random() * 1000) },
   );
-};
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {this.state.isClockVisible && <Clock name={this.state.clockName} />}
```

Comment:
Спробуй витягувати змінні через деструктуризацію, для кращої читабельності
```suggestion
  render() {
    const { isClockVisible, clockName } = this.state;

    return (
      <div className="App">
        <h1>React clock</h1>

        {isClockVisible && <Clock name={clockName} />}
```

Reply:
Погоджуюся з коментом Артура